reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -290,7 +291,10 @@ def grid_SO3_uniform(n_rotations):
             Array describing rotations.
             Shape (n_rotations, 3, 3)
         """
-        rots = np.ones((n_rotations, 3, 3))
+        geom = special_orthogonal.SpecialOrthogonal(3, "matrix")
+        rots = geom.random_uniform(n_rotations)
+        negatives = np.tile(np.random.randint(2, size=n_rotations) * 2 - 1, (3, 3, 1)).T
```

Comment:
I think you are negating every second rotation. A bit opaque in how the tiling works, so I would put a note in the docstring, or the variable name indicating how you are scaling each rotation by +1 or -1, with one random number per rotation.

Reply:
I'll add a note to the docstring. The reason for this is that geomstats only rotates on a half-sphere. So, I invert the rotations randomly in order to cover the entire sphere (I validated the functionality in a jupyter notebook and am happy to post images confirming this behaviour)

Reply:
Yes I recall this. Great work! Put as much as you can of that in the docstring and tests so that it doesn't go to waste.

The test has to be better than checking for the shape. For sure orthonormality
* orthogonality: each R.dot(R.T) is close np.eye(3)
* normality: np.det(R) is close 1

And I think something for uniformity would be great. If you can come up with some heuristic, would be a nice start. Doesn't have to be rigerous, just protest against a bad fail case. Some ideas (that work with large # of rotations)
* project to the three coordinates used to plot on sphere. Check roughly same amount above/below, through the three axes (up/down; left/right; front/back)
* convert to some encoding (euler angles???) and check stats on those encoding (I think two euler angles (ZYZ or ZXZ) are uniform and one is not)
* integrate with an even function (e.g. x**2 + y**2 + z**2) that should be double the integration on half. If the function is odd (e.g. x**3 + y**3 + z**3) it should integrate to roughly one.
  * even: (axis_3_vectors[half_idx]**2).sum() is close 2*axis_3_vectors.sum()
  * odd: (axis_3_vectors[half_idx]**3).sum() is close 0